pkg/types/types.go

```diff
@@ -197,29 +197,57 @@ type Metric struct {
 }
 
 // MergeMetrics merges metrics by name.
-func MergeMetrics(metrics [][]Metric) []Metric {
+func MergeMetrics(metrics [][]Metric, consistencyCheck bool) ([]Metric, int, int) {
 	if len(metrics) == 0 {
-		return nil
+		return nil, 0, 0
 	}
 
 	if len(metrics) == 1 {
-		return metrics[0]
+		pointCount := 0
+		ms := metrics[0]
+		for _, m := range ms {
+			pointCount += len(m.Values)
+		}
+		return ms, pointCount, 0
 	}
 
-	names := make(map[string][]Metric)
+	metricByNames := make(map[string][]Metric)
 
 	for _, ms := range metrics {
 		for _, m := range ms {
-			names[m.Name] = append(names[m.Name], m)
+			metricByNames[m.Name] = append(metricByNames[m.Name], m)
 		}
 	}
 
 	merged := make([]Metric, 0)
-	for _, ms := range names {
-		merged = append(merged, mergeMetrics(ms))
+	pointCount := 0
+	inconsistencyCount := 0
+	InconsistentMetricsLogLimit := 10
+	var limitedInconsistentMetricsForLog []map[string]interface{}
```

Comment:
nit: idiomatic go gravitates toward shorter variable names, especially when it's used only a few lines around declaration.

Reply:
hmmm. I always prefer descriptive names for the variables. It helps to know what's the purpose of that thing in advance. But I see your point. I'll try to beautify it at least :))